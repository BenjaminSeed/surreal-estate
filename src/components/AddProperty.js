import React, { useState } from "react";
import "../styles/add-property.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
      bedrooms: 0,
      bathrooms: 0,
      price: 0,
      type: "",
      email: "",
    },
  };

  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (event) => {
    event.preventDefault();
    console.log(fields);
  };

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="add-property">
      <form onSubmit={handleAddProperty}>
        <label htmlFor="title">
          Title
          <input
            type="text"
            id="title"
            name="title"
            value={fields.title}
            onChange={handleFieldChange}
          />
        </label>

        <div>
          <label htmlFor="city">
            City
            <select
              id="city"
              name="city"
              value={fields.city}
              onChange={handleFieldChange}
            >
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Shefield">Shefield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>
        </div>

        <div>
          <label htmlFor="bedrooms">
            No of Bedrooms
            <input
              type="number"
              id="bedrooms"
              name="bedrooms"
              value={fields.bedrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div>
          <label htmlFor="bathrooms">
            No of Bathrooms
            <input
              type="number"
              id="bathrooms"
              name="bathrooms"
              value={fields.bathrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div>
          <label htmlFor="price">
            Price
            <input
              type="number"
              id="price"
              name="price"
              value={fields.price}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div>
          <label htmlFor="type">
            Type
            <select
              id="type"
              name="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="Flat">Flat</option>
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-Detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Bungalow">Bungalow</option>
            </select>
          </label>
        </div>

        <div>
          <label htmlFor="email">
            Email
            <input
              type="email"
              id="email"
              name="email"
              value={fields.email}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <button type="submit">Add</button>
      </form>
      <div>Add Property Page</div>
    </div>
  );
};

export default AddProperty;
