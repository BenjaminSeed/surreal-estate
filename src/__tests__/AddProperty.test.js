import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

test("Add property", () => {
  const { asFragment } = render(<AddProperty />);
  expect(asFragment()).toMatchSnapshot();
});
