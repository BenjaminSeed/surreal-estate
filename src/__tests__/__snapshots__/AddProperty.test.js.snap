// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add property 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    <form>
      <label
        for="title"
      >
        Title
        <input
          id="title"
          name="title"
          type="text"
          value=""
        />
      </label>
      <div>
        <label
          for="city"
        >
          City
          <select
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </label>
      </div>
      <div>
        <label
          for="bedrooms"
        >
          No of Bedrooms
          <input
            id="bedrooms"
            name="bedrooms"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div>
        <label
          for="bathrooms"
        >
          No of Bathrooms
          <input
            id="bathrooms"
            name="bathrooms"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div>
        <label
          for="price"
        >
          Price
          <input
            id="price"
            name="price"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div>
        <label
          for="type"
        >
          Type
          <select
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </label>
      </div>
      <div>
        <label
          for="email"
        >
          Email
          <input
            id="email"
            name="email"
            type="email"
            value=""
          />
        </label>
      </div>
      <button
        type="submit"
      >
        Add
      </button>
    </form>
    <div>
      Add Property Page
    </div>
  </div>
</DocumentFragment>
`;
